compiler/rustc_mir/src/transform/check_consts/ops.rs

```diff
@@ -49,7 +49,9 @@ pub fn non_const<O: NonConstOp>(ccx: &ConstCx<'_, '_>, op: O, span: Span) -> boo
         return false;
     }
 
-    op.emit_error(ccx, span);
+    let mut err = op.build_error(ccx, span);
```

Comment:
The "usual" way we do this is to return an `ErrorReported` instance

Reply:
Ah, nevermind, this is for the primary/secondary system. One possibility would be not to have the `importance` function`, but instead return an `Result<ErrorReported, DiagnosticBuilder<'tcx>>` from `build_error`. Then this could internally decide whether to immediately report or whether to return the builder for later use.